sql/src/test/java/org/apache/druid/sql/calcite/schema/SystemSchemaTest.java

```diff
@@ -0,0 +1,632 @@
+        druidSchema,
+        serverView,
+        EasyMock.createStrictMock(AuthorizerMapper.class),
+        client,
+        client,
+        mapper
+    );
+  }
+
+  private final DataSegment segment1 = new DataSegment(
+      "test1",
+      Intervals.of("2010/2011"),
+      "version1",
+      null,
+      ImmutableList.of("dim1", "dim2"),
+      ImmutableList.of("met1", "met2"),
+      null,
+      1,
+      100L,
+      DataSegment.PruneLoadSpecHolder.DEFAULT
+  );
+  private final DataSegment segment2 = new DataSegment(
+      "test2",
+      Intervals.of("2011/2012"),
+      "version2",
+      null,
+      ImmutableList.of("dim1", "dim2"),
+      ImmutableList.of("met1", "met2"),
+      null,
+      1,
+      100L,
+      DataSegment.PruneLoadSpecHolder.DEFAULT
+  );
+  private final DataSegment segment3 = new DataSegment(
+      "test3",
+      Intervals.of("2012/2013"),
+      "version3",
+      null,
+      ImmutableList.of("dim1", "dim2"),
+      ImmutableList.of("met1", "met2"),
+      null,
+      1,
+      100L,
+      DataSegment.PruneLoadSpecHolder.DEFAULT
+  );
+  private final DataSegment segment4 = new DataSegment(
+      "test4",
+      Intervals.of("2017/2018"),
+      "version4",
+      null,
+      ImmutableList.of("dim1", "dim2"),
+      ImmutableList.of("met1", "met2"),
+      null,
+      1,
+      100L,
+      DataSegment.PruneLoadSpecHolder.DEFAULT
+  );
+  private final DataSegment segment5 = new DataSegment(
+      "test5",
+      Intervals.of("2017/2018"),
+      "version5",
+      null,
+      ImmutableList.of("dim1", "dim2"),
+      ImmutableList.of("met1", "met2"),
+      null,
+      1,
+      100L,
+      DataSegment.PruneLoadSpecHolder.DEFAULT
+  );
+
+  private final HttpClient httpClient = EasyMock.createMock(HttpClient.class);
+  private final DirectDruidClient client1 = new DirectDruidClient(
+      new ReflectionQueryToolChestWarehouse(),
+      QueryRunnerTestHelper.NOOP_QUERYWATCHER,
+      new DefaultObjectMapper(),
+      httpClient,
+      "http",
+      "foo",
+      new NoopServiceEmitter()
+  );
+  private final DirectDruidClient client2 = new DirectDruidClient(
+      new ReflectionQueryToolChestWarehouse(),
+      QueryRunnerTestHelper.NOOP_QUERYWATCHER,
+      new DefaultObjectMapper(),
+      httpClient,
+      "http",
+      "foo2",
+      new NoopServiceEmitter()
+  );
+  private final ImmutableDruidServer druidServer1 = new ImmutableDruidServer(
+      new DruidServerMetadata("server1", "localhost:0000", null, 5L, ServerType.REALTIME, DruidServer.DEFAULT_TIER, 0),
+      1L,
+      null,
+      ImmutableMap.of("segment1", segment1, "segment2", segment2)
+  );
+
+  private final ImmutableDruidServer druidServer2 = new ImmutableDruidServer(
+      new DruidServerMetadata("server2", "server2:1234", null, 5L, ServerType.HISTORICAL, DruidServer.DEFAULT_TIER, 0),
+      1L,
+      null,
+      ImmutableMap.of("segment2", segment2, "segment4", segment4, "segment5", segment5)
```

Comment:
Shouldn't it use `segment3`

Reply:
This is `segment2` because in the [test](https://github.com/apache/incubator-druid/pull/6094/files#diff-1b9272e9d665e89f38ead66e8435bd9dR467) later, it asserts for 2 replicas for `segment2`. So I was trying to mimic the behavior that `segment2` is served by both the servers `druidServer1` and `druidServer2`.

Reply:
It is expected that when I change the code here to `"segment3", segment3`, all tests in this class still pass?

Reply:
Tests did pass with `segment3`, I checked again and the 2 replicas for `segment2` are actually coming from this [part](https://github.com/apache/incubator-druid/pull/6094/files#diff-1b9272e9d665e89f38ead66e8435bd9dR204) , so yes it can be changed to `segment3`.Will do.

Reply:
hm, I think this suite needs a test for `SystemSchema.ServerSegmentsTable` for server/segment mappings, this test would catch that server2->server3 change

Reply:
you make a great point @jon-wei , I thought `server_segments` is not that interesting or complex, so skipped the test. Adding the test in this [PR](https://github.com/apache/incubator-druid/pull/6460). Since I switched segment2->segment3, that seems more cleaner. This new test will now catch segment3->segment2 change.